From d64b3c2b8a1657c57ace7bea5da966ca7ceb052c Mon Sep 17 00:00:00 2001
From: Zidan Wang <b50113@freescale.com>
Date: Fri, 19 Dec 2014 11:13:01 +0800
Subject: [PATCH 0329/5242] MLK-10038-1: mfd: si476x-i2c: Add support of
 si476x-rev4.0 board

commit  3e17f07060986c7529b7b2a5cbf470d75f2be484 from
https://source.codeaurora.org/external/imx/linux-imx.git

Currently, si476x-rev1.0 and si476x-rev4.0 board just support A10 compatible
command set. For si476x-rev1.0 board, its firmware revision is unsupported and
will revert to A10 compatible function. For si476x-rev4.0 board, its firmware
revision is two and will use A30 function, but A30 command set function can't
work for the rev4.0 board.
So make the command set configurable in dts. If "revision-a10" is present,
set the revision to SI476X_REVISION_A10 to use A10 compatible commit set.
Otherwise, get the revision from si476x register.

Signed-off-by: Zidan Wang <b50113@freescale.com>
(cherry picked from commit b648714c3b71ee084188ae04b1e6a6f2554fe2cb)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mfd/si476x-i2c.c |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/mfd/si476x-i2c.c b/drivers/mfd/si476x-i2c.c
index 48dc07a..b75594b 100644
--- a/drivers/mfd/si476x-i2c.c
+++ b/drivers/mfd/si476x-i2c.c
@@ -809,10 +809,14 @@ static int si476x_core_probe(struct i2c_client *client,
 	core->power_state = SI476X_POWER_UP_FULL;
 	si476x_core_set_power_state(core, SI476X_POWER_DOWN);
 
-	rval = si476x_core_get_revision_info(core);
-	if (rval < 0) {
-		rval = -ENODEV;
-		goto free_kfifo;
+	if (of_property_read_bool(client->dev.of_node, "revision-a10"))
+		core->revision = SI476X_REVISION_A10;
+	else {
+		rval = si476x_core_get_revision_info(core);
+		if (rval < 0) {
+			rval = -ENODEV;
+			goto free_kfifo;
+		}
 	}
 
 	cell_num = 0;
-- 
1.7.9.5

